File: packages/youtube-player/android/src/main/java/io/capawesome/capacitorjs/plugins/youtubeplayer/classes/events/CurrentTimeChangeEvent.java
package io.capawesome.capacitorjs.plugins.youtubeplayer.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.youtubeplayer.interfaces.Result;

public class CurrentTimeChangeEvent implements Result {

    private final double currentTime;

    @NonNull
    private final String id;

    public CurrentTimeChangeEvent(double currentTime, @NonNull String id) {
        this.currentTime = currentTime;
        this.id = id;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("currentTime", currentTime);
        result.put("id", id);
        return result;
    }
}

File: packages/youtube-player/ios/Plugin/Classes/Events/CurrentTimeChangeEvent.swift
import Capacitor
import Foundation

@objc public class CurrentTimeChangeEvent: NSObject, Result {
    let currentTime: Double
    let id: String

    init(currentTime: Double, id: String) {
        self.currentTime = currentTime
        self.id = id
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["currentTime"] = currentTime
        result["id"] = id
        return result as AnyObject
    }
}
